geonode/security/models.py

```diff
@@ -388,3 +388,44 @@ def set_workflow_perms(self, approved=False, published=False):
             if settings.OGC_SERVER['default'].get("GEOFENCE_SECURITY_ENABLED", False):
                 if self.polymorphic_ctype.name == 'layer':
                     sync_geofence_with_guardian(self.layer, VIEW_PERMISSIONS)
+
+    def get_user_perms(self, user):
+        """
+        Returns a list of permissions a user has on a given resource
+        """
+        resource = self.get_self_resource()
+        user_resource_perms = get_users_with_perms(resource).get(user, [])
+        try:
+            if hasattr(self, "layer"):
+                user_resource_perms.extend(
+                    get_users_with_perms(self.layer).get(user, [])
+                    )
+        except Exception:
+            logger.debug(traceback.format_exc())
+
+        return user_resource_perms
+
+    def user_can(self, user, permission):
+        """
+        Checks if a has a given permission to the resource
+        """
+        # To avoid circular import
+        from geonode.base.models import Configuration
+
+        config = Configuration.load()
```

Comment:
@maabdelghaffar if read-only mode is active this should return False ONLY for edit permissions.